src/main/java/com/coyoapp/tinytask/repository/DefaultFileRepository.java

```diff
@@ -0,0 +1,50 @@
+package com.coyoapp.tinytask.repository;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+@Component
+public class DefaultFileRepository implements FileRepository {
+
+  @Value("${file.base.dir}")
+  private String filesBaseDir;
+
```

Comment:
I would simplify and rename the method like below.

```
  public Path createFileBaseDirectoryIfNotExists() {
    try {
      return Files.createDirectories(Paths.get(filesBaseDir));
    } catch (IOException e) {
      throw new IllegalArgumentException("Could not create file base dir", e);
    }
  }
```

this optional makes the code unnecessarily complicated. especially an exception is thrown anyway if the directory could not be created (see DefaultFileService.initImageRootFolder())